internal/process/steps/gardener_cluster.go

```diff
@@ -16,13 +17,69 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func NewSyncGardenerCluster(os storage.Operations, k8sClient client.Client) syncGardenerCluster {
-	return syncGardenerCluster{
+const GardenerClusterStateReady = "Ready"
+
+func NewSyncGardenerCluster(os storage.Operations, k8sClient client.Client) *syncGardenerCluster {
+	return &syncGardenerCluster{
+		k8sClient:        k8sClient,
+		operationManager: process.NewOperationManager(os),
+	}
+}
+
+func NewCheckGardenerCluster(os storage.Operations, k8sClient client.Client) *checkGardenerCluster {
+	return &checkGardenerCluster{
 		k8sClient:        k8sClient,
 		operationManager: process.NewOperationManager(os),
 	}
 }
 
+type checkGardenerCluster struct {
+	k8sClient        client.Client
+	operationManager *process.OperationManager
+}
+
+func (_ *checkGardenerCluster) Name() string {
+	return "Check_GardenerCluster"
+}
+
+func (s *checkGardenerCluster) Run(operation internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	gc, err := s.GetGardenerCluster(operation.RuntimeID, operation.KymaResourceNamespace)
+	if err != nil {
+		log.Errorf("unable to get gardener cluster %s/%s", operation.KymaResourceNamespace, operation.RuntimeID)
+		return s.operationManager.RetryOperation(operation, "unable to get gardener cluster", err, time.Second, 10*time.Second, log)
```

Comment:
By the way I am still wondering if usage of term `Gardener cluster resource` would be more informative.